lib/getStagedFiles.js

```diff
@@ -4,8 +4,14 @@ const execGit = require('./execGit')
 
 module.exports = async function getStagedFiles(options) {
   try {
-    const lines = await execGit(['diff', '--staged', '--diff-filter=ACMR', '--name-only'], options)
-    return lines ? lines.split('\n') : []
+    // Docs for --diff-filter option: https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---diff-filterACDMRTUXB82308203
+    // Docs for -z option: https://git-scm.com/docs/git-diff#Documentation/git-diff.txt--z
+    const lines = await execGit(
+      ['diff', '--staged', '--diff-filter=ACMR', '--name-only', '-z'],
+      options
+    )
+    // eslint-disable-next-line no-control-regex
+    return lines ? lines.replace(/\u0000$/, '').split('\u0000') : []
```

Comment:
Should this replace be global and multiline? I assume this will be on every line.

Reply:
Ah, true, you are right about `/g`! What do you mean about it being `multiline`?

Reply:
Oh wait actually the replace here is for removing a trailing char. No need for `/g`

Reply:
With `-z`, git prints `fileA\u0000fileB\u0000fileC\u0000`. We need to remove the last occurrence of `\u0000` before splitting.

Reply:
Ok, cool. Maybe add a comment here to explain why replace is done.

Reply:
✅ done